charts/kube-prometheus-stack/templates/grafana/crd-datasources.yaml

```
@@ -0,0 +1,27 @@
+{{- if or (and .Values.grafana.enabled .Values.grafana.sidecar.datasources.enabled) .Values.grafana.forceDeployDatasources }}
+{{- if .Values.grafana.operatorDatasources }}
+{{- $datasources := fromYaml (include "kube-prometheus-stack.grafana.datasources" .) }}
+{{- range $ds := $datasources.datasources }}
+---
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDatasource
+metadata:
+  name: {{ include "kube-prometheus-stack.fullname" $ }}-{{ include "kube-prometheus-stack.grafana.sanitizeName" $ds.name }}
+  namespace: {{ template "kube-prometheus-stack-grafana.namespace" $ }}
+  labels:
+    app: {{ template "kube-prometheus-stack.name" $ }}-grafana
+    {{ include "kube-prometheus-stack.labels" $ | nindent 4 }}
+spec:
+  instanceSelector:
+    matchLabels:
+      {{- if $.Values.grafana.operatorInstanceSelector }}
+      {{- toYaml $.Values.grafana.operatorInstanceSelector | nindent 6 }}
+      {{- else }}
+      app.kubernetes.io/instance: grafana
+      app.kubernetes.io/name: grafana-operator
+      {{- end }}
```

To reduce the complexity and hidden defaults, what about:

```suggestion
    matchLabels:
      {{- toYaml $.Values.grafana.operatorInstanceSelector | nindent 6 }}
```

Are they any downsides using this + the empty map on the values.yaml? I'm not a export into the operator.